File: leetcode/分组刷题/1数组/2双指针/11盛最多水的容器.py
# -*- coding: utf-8 -*-
"""
Created on Mon May 27 21:12:59 2019

@author: 鱼红叶
"""

class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        
        max_ = 0
        i, j = 0, len(height)-1
        while i < j:
            max_ = max(max_, (j-i) * min(height[i], height[j]))
            if height[i] < height[j] :
                i += 1
            else:
                j -= 1
        return max_
    
print(Solution().maxArea([1,8,6,2,5,4,8,3,7]))

File: leetcode_2021/leetcode_cpp/11.盛最多水的容器.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0;
        int j = height.size() - 1;
        int max_area = 0;
        while (i < j) {
            max_area = max(max_area, (j-i) * min(height[i], height[j]));
            if (height[i] < height[j]) {
                i += 1;
            } else {
                j -= 1;
            }
        }
        return max_area;
    }
};